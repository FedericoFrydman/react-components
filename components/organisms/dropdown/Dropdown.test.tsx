import { shallow, ShallowWrapper } from 'enzyme';
import {
  Dropdown,
  DropdownCheckboxItem,
  DropdownItem,
  DropdownItemWithSubMenu,
  DropdownLabel,
  DropdownRadioGroup,
  DropdownRadioItem,
  DropdownSeparator,
} from './Dropdown';
import { Fragment } from 'react';

describe('Dropdown Component', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    const openAlert = (ev: React.MouseEvent<HTMLDivElement>) => {
      ev.preventDefault();
      alert('Action button pressed');
    };

    wrapper = shallow(
      <Dropdown.Root>
        <Dropdown.Trigger />
        <Dropdown.Portal>
          <DropdownLabel>Actions</DropdownLabel>
          <DropdownItem onClick={openAlert}>
            New Tab
            <div className="right-slot">⌘+T</div>
          </DropdownItem>
          <DropdownItem onClick={openAlert}>
            New Window
            <div className="right-slot">⌘+N</div>
          </DropdownItem>
          <DropdownItem disabled>New Private Window</DropdownItem>

          <DropdownSeparator />
          <DropdownLabel>Preferences</DropdownLabel>
          <DropdownCheckboxItem checked={false} onCheckedChange={() => null}>
            Show Bookmarks <div className="right-slot">⌘+B</div>
          </DropdownCheckboxItem>
          <DropdownCheckboxItem checked={true} onCheckedChange={() => null}>
            Show Full URLs
          </DropdownCheckboxItem>

          <DropdownSeparator />

          <DropdownLabel>Radio Buttons</DropdownLabel>
          <DropdownRadioGroup value="one" onValueChange={() => null}>
            <DropdownRadioItem value="one">Option One</DropdownRadioItem>
            <DropdownRadioItem value="two">Option Two</DropdownRadioItem>
          </DropdownRadioGroup>
          <DropdownSeparator />
          <DropdownItemWithSubMenu
            subMenu={
              <Fragment>
                <DropdownItem>Create Shortcut…</DropdownItem>
                <DropdownItem>Name Window…</DropdownItem>
                <DropdownSeparator />
                <DropdownItem>Developer Tools</DropdownItem>
              </Fragment>
            }
          >
            Demo
          </DropdownItemWithSubMenu>
        </Dropdown.Portal>
      </Dropdown.Root>,
    );
  });

  test('should render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
