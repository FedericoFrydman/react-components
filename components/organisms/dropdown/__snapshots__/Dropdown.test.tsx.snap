// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component should render the component 1`] = `
<DropdownMenu>
  <DropdownTrigger />
  <DropdownPortal>
    <DropdownLabel>
      Actions
    </DropdownLabel>
    <DropdownItem
      onClick={[Function]}
    >
      New Tab
      <div
        className="right-slot"
      >
        ⌘+T
      </div>
    </DropdownItem>
    <DropdownItem
      onClick={[Function]}
    >
      New Window
      <div
        className="right-slot"
      >
        ⌘+N
      </div>
    </DropdownItem>
    <DropdownItem
      disabled={true}
    >
      New Private Window
    </DropdownItem>
    <DropdownSeparator />
    <DropdownLabel>
      Preferences
    </DropdownLabel>
    <DropdownCheckboxItem
      checked={false}
      onCheckedChange={[Function]}
    >
      Show Bookmarks 
      <div
        className="right-slot"
      >
        ⌘+B
      </div>
    </DropdownCheckboxItem>
    <DropdownCheckboxItem
      checked={true}
      onCheckedChange={[Function]}
    >
      Show Full URLs
    </DropdownCheckboxItem>
    <DropdownSeparator />
    <DropdownLabel>
      Radio Buttons
    </DropdownLabel>
    <DropdownRadioGroup
      onValueChange={[Function]}
      value="one"
    >
      <DropdownRadioItem
        value="one"
      >
        Option One
      </DropdownRadioItem>
      <DropdownRadioItem
        value="two"
      >
        Option Two
      </DropdownRadioItem>
    </DropdownRadioGroup>
    <DropdownSeparator />
    <DropdownItemWithSubMenu
      subMenu={
        <React.Fragment>
          <DropdownItem>
            Create Shortcut…
          </DropdownItem>
          <DropdownItem>
            Name Window…
          </DropdownItem>
          <DropdownSeparator />
          <DropdownItem>
            Developer Tools
          </DropdownItem>
        </React.Fragment>
      }
    >
      Demo
    </DropdownItemWithSubMenu>
  </DropdownPortal>
</DropdownMenu>
`;
